neuralhydrology/utils/config.py

```diff
@@ -544,6 +552,38 @@ def save_weights_every(self) -> int:
     def seed(self) -> int:
         return self._cfg.get("seed", None)
 
+    @property
+    def transformer_nlayers(self) -> int:
+        return self._cfg.get("transformer_nlayers", None)
+
+    @property
+    def transformer_positional_encoding_type(self) -> str:
+        return self._cfg.get("transformer_positional_encoding_type", None)
+
+    @property
+    def transformer_embedding_dimension(self) -> int:
+        return self._cfg.get("transformer_embedding_dimension", None)
+
+    @property
+    def transformer_activation(self) -> str:
+        return self._cfg.get("transformer_activation", None)
+
+    @property
+    def transformer_dim_feedforward(self) -> int:
+        return self._cfg.get("transformer_dim_feedforward", None)
+
+    @property
+    def transformer_positional_dropout(self) -> int:
+        return self._cfg.get("transformer_positional_dropout", None)
+
+    @property
+    def transformer_dropout(self) -> float:
+        return self._cfg.get("transformer_dropout", None)
+
+    @property
+    def transformer_nheads(self) -> float:
+        return self._cfg.get("transformer_nheads", None)
+
```

Comment:
I didn't check carefully but I think all of these arguments are hard requirements for the transformer model, no? In that case we use `return self._get_value_verbose(key)` as this will raise an error if one of this properties is used somewhere, but it was not specified in the config. Also, I think your defaults do not really make sense. E.g. `transformer_nlayers` is required. If not specified in the config, your code would return `None` and then a PyTorch error is raised, because `None` is not a valid input value for PyTorchs TransformerEncoder. It is also okay to have defaults if one argument is not set in the config, but the only place where they should be set is here (and not in the model with something like `ìf cfg.transformer_nlayers is None: # use 1 layer`). So maybe have another look over the defaults here. 
Also, if I am not wrong `transformer_activation` is never used.